File: problem0003/3.c
#include<stdio.h>
#define TRUE 1
#define FALSE 0
int IsPrime(int x);
int main()
{
    unsigned long long num = 600851475143;
    int x;
 
    for (x=2;num>1;x++)
	{
        if (IsPrime(x) && (num % x == 0))
        {
            num = num / x;
        }
	}
    printf("%d\n", x);
    return 0;
}
 
int IsPrime(int x)
{
    int y;
    for (y=3;y<x;y++)
        if (y % x == 0)
            return FALSE;

    return TRUE;
}

File: problem0003/3.py
def isprime(x):
    for y in range(x-1,1,-1):
        if x % y == 0:
            return False
    return True

num = 600851475143
for x in range(1,7000,2):
    prime = x if isprime(x) else None
    if not prime is None and num % prime == 0:
        num = (num / prime)
    if num <= 1:
        break

print x
